### PR TITLE
Add Workflow Executor Example

### PR DESCRIPTION
**Pull request by JoshuaL3000**
## Description

Add workflow executor example using existing react-langgraph strategy.

## Issues

RFC: https://github.com/opea-project/docs/pull/77/files

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds new functionality)

### CODE REVIEW

File: WorkflowExecAgent/docker_compose/docker_compose.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    container_name: supervisor-agent-endpoint
+    volumes:
+      - ${WORKDIR}/GenAIComps/comps/agent/langchain/:/home/user/comps/agent/langchain/
+      - ${TOOLSET_PATH}:/home/user/tools/
+    ports:
+      - "9090:9090"
+    ipc: host
+    environment:
+      ip_address: ${ip_address}
+      strategy: workflow_executor
```
Comment (minmin-intel) [inline WorkflowExecAgent/docker_compose/docker_compose.yaml:16]: My recommendation is to instantiate supervisor agent and the worker agents as separate react agents the existing react_langgraph or react_llama strategy in the agent component. Each worker agent can be registered as a tool for the supervisor agent in the hierarchical architecture that we currently support. My recommendation is not to add this new "workflow_executor" strategy. We just need to support custom prompt templates, which is already planned for v1.1 release.

**Review by minmin-intel — CHANGES_REQUESTED** [review]

File: WorkflowExecAgent/docker_compose/docker_compose.yaml (hunk 1 above)
Comment (JoshuaL3000) [reply WorkflowExecAgent/docker_compose/docker_compose.yaml]: Updated strategy to be a single-tool, single react-langgraph

**Comment by JoshuaL3000** [comment]
Hi @xuechendi, the CICD validation test can only pass after the addition of custom system prompts and the merge for https://github.com/opea-project/GenAIComps/pull/782

**Review by minmin-intel — APPROVED** [review]

**Review by xuechendi — APPROVED** [review]
LGTM

**Comment by xuechendi** [comment]
@JoshuaL3000 , codes looks good to me, I was discussing with Minmin on your previous WorkExecutor strategy. Once this example is merged, we can't continue on that strategy

File: WorkflowExecAgent/docker_compose/docker_compose.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2024 Intel Corporation
```
Comment (chensuyue) [inline WorkflowExecAgent/docker_compose/docker_compose.yaml:1]: Please follow this code structure:

**Comment by JoshuaL3000** [comment]
Hi @xuechendi @minmin-intel, the CICD results are passing. Could you help to review those and see if this is ready to be merged? thanks

**Comment by xuechendi** [comment]
Result is looking good
image

**Review by lkk12014402 — APPROVED** [review]

**Pull request merged**